semgrep/semgrep/join_rule.py

```diff
@@ -0,0 +1,404 @@
+import json
+import logging
+import sys
+import tempfile
+from collections import defaultdict
+from enum import Enum
+from functools import reduce
+from itertools import chain
+from pathlib import Path
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Set
+from typing import Tuple
+from typing import Type
+
+import attr
+import peewee as pw
+from peewee import ModelSelect
+from ruamel.yaml import YAML
+
+import semgrep.semgrep_main
+from semgrep.config_resolver import Config
+from semgrep.config_resolver import resolve_config
+from semgrep.error import ERROR_MAP
+from semgrep.error import FATAL_EXIT_CODE
+from semgrep.error import Level
+from semgrep.error import SemgrepError
+from semgrep.pattern_match import PatternMatch
+from semgrep.rule import Rule
+from semgrep.rule_match import RuleMatch
+
+yaml = YAML()
+
+logger = logging.getLogger(__file__)
```

Comment:
Let's reuse the same loggers the rest of the code uses: https://github.com/returntocorp/semgrep/blob/develop/semgrep/semgrep/core_runner.py#L43 and instantiated two lines below that